### ISSUE TITLE
Creating guidelines for Reviewer Discretion for merging PRs that do not conform accurately to the Style Guide

### ISSUE THREAD

**Issue by Babapool** [kubernetes/website#40934]
**This is a Feature Request**

**What would you like to be added**

A guide/handbook that PR reviewers can use to determine how much discreet they can be to allow merging of important PRs (like the brand new docs for a new release) having lax style rules without compromising accuracy .

**Why is this needed**

In the previous bi-weekly [SIG Docs Zoom Meeting](https://docs.google.com/document/d/1emuO4nmaQq3K8JZ9-MQeIygtrCPO9kWv7U7RzTaW4F8/), Tim pointed that SIG Docs wants new documentation to follow the style guide very accurately. However it was observed that the previous release (v1.27) of Kubernetes,  documentation doesn't follow the style guide as accurately as desired. 

A solution to this could be the merging PRs with lax styling guide only after it is properly communicated that a follow up issue/PR should be raised for the same. Such follow up issues can be "good first issues" that new contributors can work on.

Since the discretion can vary from person to person, there is a need to establish clear guidelines on how much laxity can be allowed for style guide adherence without compromising on the accuracy of the docs. A guide/handbook will document the type of flexibility that can be allowed to merge such PRs ensuring uniformity and the steps to be taken later to ensure that a follow up PR is raised.

**Comments**

While making such guidelines, we should ensure one laxity like machine translations should not be missed. [See](https://developers.google.com/style/spelling) for more information.

**Comment by sftim** [comment]
The document you'd like to create, would it look a bit like https://kubernetes.io/docs/contribute/review/for-approvers/?

**Comment by sftim** [comment]
(essentially: we already have something)

Improvements and suggestions are welcome.

**Comment by sftim** [comment]
There is also https://kubernetes.io/docs/contribute/review/reviewing-prs/#language-and-grammar

**Comment by Babapool** [comment]
I personally think that the document to be create should look more like https://kubernetes.io/docs/contribute/review/reviewing-prs/#language-and-grammar. Thoughts, what do other contributors think.

**Comment by Babapool** [comment]
>(essentially: we already have something)
>Improvements and suggestions are welcome.

I am feel like if there are following things in the PR, the PR should be allowed to merge/slide (  _provided **a follow up issue/PR is raised soon and the PR actually improves the docs and is technically accurate**_ ) :

- A handful of spelling mistakes (max 4-5). These spellings mistakes should not be of technical terms like "Kubernetes", and other "API objects.
- If the formatting (like indentation, bulleting, line wrapping) of the docs does not hamper the readers ability to read and understand the documentation.
- Some inconsistencies (not more than 4-5 in number) in capitalization or punctuation, that do not significantly impact technical accuracy and understandability of the documentation.

I would be great to hear from other reviewers and contributors their views on this issue.

**Comment by tengqm** [comment]
The typos, mistakes, indentation, bulleting, line wrapping, inconsistencies are all things we as reviewers should take care of. What does 'lax style rules' buy us if we decide to lower the bar for PR merges?

It is probably okay if 1) a PR is very urgent to be merged before a certain date; 2) there are known issues in the PR as is, but the contributor has promised a follow-up to fix them; 3) the PR itself is already very large to incorporate more changes. There are also cases where a contributor simply doesn't want to accept feedbacks from others and these cases are not rare.

Let't think about these again:

1. A PR is urgent ... okay, why the urgency? A release-notes can be urgent because it is supposed to be out of the door tomorrow. We can fix the known issues later. Any other scenarios we can think of?
2. The contributor promised a follow-up ... well, most of the time, I didn't see that happening. Many of our contributors are multi-tasked. They forget to come back to a PR that is already merged.
3. A PR is too large to be further revised/rebased, ... Emm ... why do we want such a PR anyway? Maybe we should instead close the large PR and split it into several smaller ones for ease of review.

**Comment by sftim** [comment]
/triage accepted

We should improve guidance in this area.
